src/lib/Dungeon.js

```diff
@@ -415,6 +465,38 @@ class AutomationDungeon
         }
     }
 
+    /**
+     * @brief Returns the cell with the most non-visible neighbors
+     *
+     * @param cellsToChooseFrom: The list of cells to analyze
+     *
+     * @param unwrapedTiles: The preprocessed tiles (for faster access)
+     */
+    static __internal__getCellWithMostNonVisibleNeightbours(cellsToChooseFrom, unwrapedTiles)
+    {
+        const cellsWithNeightbours = cellsToChooseFrom.map(({ tile, i, j }) => {
+            const nonVisibleNeighborsCount = [
+                unwrapedTiles[`${i - 1},${j - 1}`],
+                unwrapedTiles[`${i - 1},${j}`],
+                unwrapedTiles[`${i - 1},${j + 1}`],
+                unwrapedTiles[`${i},${j - 1}`],
+                unwrapedTiles[`${i},${j + 1}`],
+                unwrapedTiles[`${i + 1},${j - 1}`],
+                unwrapedTiles[`${i + 1},${j}`],
+                unwrapedTiles[`${i + 1},${j + 1}`],
+            ].filter(neigh => neigh && !neigh.tile._isVisible).length;
+            return { tile, i, j, nonVisibleNeighborsCount };
+        });
```

Comment:
I found this difficult to read and understand, I'm not familial with this syntaxe, is there a more readable way to write this ?

Reply:
Maybe add a function `__internal_getNeighbors` and instead of building the array in-place
in the function build an array an use a loop to insert elements to it.

I'm also thinking an object might not be the best choice for `unwrapedTiles`
Instead of building it this way, couldn't you use a two-dimensional array instead ?